### ISSUE TITLE
Migrate library jackc/pgx from v3 to v4 (v3 doesn't work on windows)

### ISSUE THREAD

**Issue by ilkhenrique** [kingluo/pgcat#1]
Thank you very much for creating pgcat. It's fantastic.

**Please** migrate jackc/pgx from v3 to v4!
There are so many important fixes since v3. Also v3 does not work properly on windows.

Thank you again,
Dennis

**Comment by kingluo** [comment]
I would try to update pgx to v4. It would take me some time because v4 is breakage change from v3.

> Also v3 does not work properly on windows.

I tested pgcat on Windows, and it seems that v3  works on Windows.
Could you clarify more where v3 does not work properly on Windows?

**Comment by kingluo** [comment]
I commit my changes, please have a check.

**Comment by ilkhenrique** [comment]
Thank you very much!! 
I will check.

v3 seems to work but it does not. The problem is the way windows winsock return codes and when it returns. 
Actually I made some workaround on pgx v3 to make it work, but not really fixes, just bypassed problematic situations.

Here the problem is reported:
https://gitmemory.com/issue/jackc/pgx/637/554683007

But it's said to have been fixed on v4 somewhere else
Thank you again!
